CMakeLists.txt

```diff
@@ -471,7 +471,11 @@ enable_testing() # Enable for tests without binary
 
 option(ENABLE_OPENSSL "This option performs a build with OpenSSL. NOTE! This option is insecure and should never be used. By default, ClickHouse uses and only supports BoringSSL" OFF)
 
-option(ENABLE_OPENSSL_DYNAMIC "This option removes SSL from ClickHouse and will link to the OpenSSL version supplied by OS." OFF)
+if(ARCH_S390X)
```

Comment:
Instead of l. 474, please use something like:
```cmake
if (ARCH_S390X)
    set(ENABLE_OPENSSL_DYNAMIC_DEFAULT, ON)
else ()
    set(ENABLE_OPENSSL_DYNAMIC_DEFAULT, OFF)
endif ()
option(ENABLE_OPENSSL_DYNAMIC "This option removes SSL from ClickHouse and will link to the OpenSSL version supplied by OS." ${ENABLE_OPENSSL_DYNAMIC_DEFAULT})
```